tests/test_predictors.py

```
@@ -1252,83 +1318,123 @@ def test_search_batch(self, single_label_index, normalized_vectors):
+                assert isinstance(label, tuple), f"Expected tuple, got {type(label)}"
+
+    def test_dump_and_load_mixed_scalar_and_tuple_labels(
+        self, normalized_vectors, tmp_path
+    ):
+        """A mix of scalar and tuple labels round-trips correctly."""
+        labels = ["A", ("B", "C"), 42, ("D",), "E", 99]
+        index = VectorIndex(vectors=normalized_vectors, labels=labels)
+        index.dump(str(tmp_path))
+        loaded = load_vector_index(str(tmp_path))
+        assert loaded.labels == labels
+        for orig, roundtripped in zip(labels, loaded.labels):
```

Ruff (0.15.2)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)